sscanss/app/dialogs/preferences.py

```diff
@@ -249,6 +249,36 @@ def generalForm(self):
         frame.setLayout(main_layout)
         self.stack.addWidget(create_scroll_area(frame))
 
+    def renderingSizeSlider(self, key, value, range=(5, 100)):
+        """Creates a horizontal layout containing a slider 
+            and editable value for a graphics rendering size
+        
+            :param key: a string representing a settings key
+            :type key: str
+            :param value: integer rendering size
+            :type value: int       
+            :param range: min and max values the slider can take
+            :type range: tuple
+            """
+        layout = QtWidgets.QHBoxLayout()
+
+        slider = QtWidgets.QSlider(orientation=QtCore.Qt.Orientation.Horizontal)
+        slider.setRange(range[0], range[1])
+        slider.setProperty(self.prop_name, (key, value))
+        slider.setValue(slider.property(self.prop_name)[1])
+
+        slider_value = QtWidgets.QLineEdit(str(slider.value()))
+        slider_value.setFixedWidth(40)
+        slider_value.textChanged.connect(lambda: slider.setValue(int(slider_value.text()))
+                                         if slider_value.text().isnumeric() else 0)
+        slider.valueChanged.connect(lambda: slider_value.setText(str(slider.value())))
+        slider.valueChanged.connect(lambda: self.changeSetting(slider.value()))
+
+        layout.addWidget(slider)
+        layout.addWidget(slider_value)
+
+        return layout
+
     def graphicsForm(self):
         """Creates form inputs for graphics settings"""
         self.addGroup(settings.Group.Graphics)
```

Comment:
Please add some spacing to the graphicsForm i.e. `main_layout.setSpacing(10)`

Reply:
I agree with this added spacing as an improvement, although I'd suggest that we additionally increase the preferences window minimum height from 480 -> 520 to allow the user to see all the various settings by default.